src/app/clients/form/submitFormHelpers.ts

```diff
@@ -274,34 +62,50 @@ const formatClientRecord = (
     };
 };
 
-export const submitAddClientForm: SubmitFormHelper = async (fields, router) => {
+export const submitAddClientForm = async (
+    fields: ClientFields,
+    router: AppRouterInstance
+): Promise<void> => {
     const clientRecord = formatClientRecord(fields);
-    const ids = await insertClient(clientRecord);
-    try {
-        await insertFamily([...fields.adults, ...fields.children], ids.family_id);
-        router.push(`/parcels/add/${ids.primary_key}`);
-    } catch (error) {
-        await revertClientInsert(ids.primary_key);
-        const logId = await logErrorReturnLogId(`Error with insert: Client ID ${ids.primary_key}`);
-        throw new DatabaseError("update", "client", logId);
+    const familyMembers = getFamilyMembers(fields.adults, fields.children);
+    const { data: clientId, error } = await supabase.rpc("insertClientAndTheirFamily", {
+        clientrecord: clientRecord,
+        familymembers: familyMembers,
+    });
+    if (error) {
+        const logId = await logErrorReturnLogId(
+            "Error with inserting new client and their family",
+            {
+                error,
+            }
+        );
+
+        throw new DatabaseError("insert", "client", logId);
     }
+    router.push(`/parcels/add/${clientId}`);
 };
 
-export const submitEditClientForm: SubmitFormHelper = async (
-    fields,
-    router,
-    initialFields,
-    primaryKey
-) => {
+export const submitEditClientForm = async (
+    fields: ClientFields,
+    router: AppRouterInstance,
+    primaryKey: string
+): Promise<void> => {
     const clientRecord = formatClientRecord(fields);
-    const clientBeforeUpdate = formatClientRecord(initialFields!);
-    const ids = await updateClient(clientRecord, primaryKey!);
-    try {
-        await updateFamily(fields.adults, fields.children, ids.family_id);
-        router.push(`/clients?clientId=${primaryKey}`);
-    } catch (error) {
-        await revertClientUpdate(clientBeforeUpdate, primaryKey!);
-        const logId = await logErrorReturnLogId(`Error with update: Client ID ${ids.primary_key}`);
+    const familyMembers = getFamilyMembers(fields.adults, fields.children);
+    const { data: clientId, error } = await supabase.rpc("updateClientAndTheirFamily", {
+        clientrecord: clientRecord,
+        familymembers: familyMembers,
+        clientid: primaryKey,
+    });
+    if (error) {
+        const logId = await logErrorReturnLogId(
+            `Error with updating client and their family: Client id ${primaryKey}`,
+            {
+                error,
+            }
+        );
+
         throw new DatabaseError("update", "client", logId);
     }
+    router.push(`/parcels/add/${clientId}`);
```

Comment:
Can we redirect to `router.push(`/clients?clientId=${primaryKey}`);` as before please?